blog-posts/en/angular-http-only-cookie.md

```diff
@@ -0,0 +1,423 @@
+
+First of all, your backend project and your Angular project must be located in the same domain. For example, if your Angular project is located in `https://localhost:44300`. Then your backend project `*.Host` must be located in `https://localhost:44300/api`.
+
+## Step 1: Configure your `*Web.Core` project
+
+### Add HttpOnly cookies to `Authenticate` method
+
+Find the `Authenticate` method and add the above code after the `var accessToken = CreateAccessToken( ...` line. And also you need to add refresh token cookie as well. Your code should look like this:
+
+```csharp
+// Other codes ...
+
+    var accessToken = CreateAccessToken(
+        await CreateJwtClaims(
+            loginResult.Identity,
+            loginResult.User,
+            refreshTokenKey: refreshToken.key
+        )
+    );
+
+    Response.Cookies.Append("Abp.AuthRefreshToken",
+        refreshToken.token,
+        new CookieOptions
+        {
+            Expires = Clock.Now.Add(_configuration.RefreshTokenExpiration),
+            HttpOnly = true,
+            Secure = true,
+            SameSite = SameSiteMode.None
+        }
+    );
+
+    Response.Cookies.Append("Abp.AuthToken",
+        accessToken,
+        new CookieOptions
+        {
+            Expires = Clock.Now.Add(_configuration.AccessTokenExpiration),
+            HttpOnly = true,
+            Secure = true,
+            SameSite = SameSiteMode.None
+        }
+    );
+
+    return new AuthenticateResultModel
+    {
+        AccessToken = accessToken,
+        ExpireInSeconds = (int) TimeSpan.FromSeconds(15).TotalSeconds,
+        RefreshToken = refreshToken.token,
+        RefreshTokenExpireInSeconds = (int) _configuration.RefreshTokenExpiration.TotalSeconds,
+        EncryptedAccessToken = GetEncryptedAccessToken(accessToken),
+        TwoFactorRememberClientToken = twoFactorRememberClientToken,
+        UserId = loginResult.User.Id,
+        ReturnUrl = returnUrl
+    };
+
+}
+```
+
+We need to add `Abp.AuthToken` cookie to `ImpersonatedAuthenticate`, `DelegatedImpersonatedAuthenticate`, `LinkedAccountAuthenticate` and `ExternalAuthenticate` methods in the same way as we added above. 
+
+### Add HttpOnly cookies to `RefreshToken` method
+
+We need to add get refresh token from cookie and generate new access token with it. Then, we need to add access token cookie. 
+
+* First of all, get refresh token from cookie instead of function parameter.
+
+* Then, add access token to cookie after creating access token.
+
+After updating the `RefreshToken` method, your code should look like this:
+
+```csharp
+[HttpPost]
+public async Task<RefreshTokenResult> RefreshToken()
+{
+    // get refresh token from cookie
+    var refreshToken = Request.Cookies["Abp.AuthRefreshToken"];
+    
+    if (string.IsNullOrWhiteSpace(refreshToken))
```

Comment:
Don't know if you've seen this, but when I run the `refreshToken` is null at this point, right after successfully authenticating. I _think_ I've made all the changes in all the right places.

Reply:
We get such an error when we first open the login page (index.html). It's related to redirect progress. You need to set the home page `/account/login`. Unfortunately, we do not have a better solution at the moment.

Reply:
Thanks for that info!